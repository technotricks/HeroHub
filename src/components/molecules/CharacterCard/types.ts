import {ItemClickCallback} from '@/types/base';
import {Character} from '@/types/character';

export type CardProps = {
  item: Character;
  onPress: ItemClickCallback;
};
