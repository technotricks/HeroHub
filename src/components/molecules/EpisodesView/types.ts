import {CharacterDetail} from '@/types/character';

export type EpisodesViewProps = {
  item?: CharacterDetail;
};
