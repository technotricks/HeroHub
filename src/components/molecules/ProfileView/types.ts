import {CharacterDetail} from '@/types/character';

export type ProfileViewProps = {
  item?: CharacterDetail;
};
