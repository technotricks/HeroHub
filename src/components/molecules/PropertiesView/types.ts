import {CharacterDetail} from '@/types/character';

export type PropertiesViewProps = {
  item?: CharacterDetail;
};
