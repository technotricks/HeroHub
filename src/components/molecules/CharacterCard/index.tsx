import {
  CharacterImageSmall,
  ItemContainer,
  ItemText,
  HText,
} from '@/components/atom';
import {TouchableOpacity} from 'react-native';
import {CardProps} from './types';

const CharacterCard: React.FC<CardProps> = ({item, onPress}) => (
  <ItemContainer key={item.id}>
    <TouchableOpacity
      testID={'buttonID' + item.id}
      onPress={() => onPress(item)}>
      <CharacterImageSmall
        testID={'imageID' + item.id}
        source={{uri: item.image}}
      />
      <HText
        testID={`nameID${item.id}`}
        color="#FF87A2"
        aligin="center"
        isBold
        fontSize="18px">
        {item.name}
      </HText>

      <HText
        testID={'statusID' + item.id}
        color={item?.status == 'Alive' ? '#2CB9B0' : '#FF3600'}
        aligin="left"
        isBold
        fontSize="16px">
        {item?.status !== 'unknown' ? item?.status : ''}
      </HText>
    </TouchableOpacity>
  </ItemContainer>
);

export default CharacterCard;
