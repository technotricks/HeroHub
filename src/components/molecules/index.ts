export {default as CharacterCard} from './CharacterCard';
export {default as ProfileView} from './ProfileView';
export {default as PropertiesView} from './PropertiesView';
export {default as EpisodesView} from './EpisodesView';
