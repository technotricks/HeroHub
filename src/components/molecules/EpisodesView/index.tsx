import {HeadingContainer, Cell, Row, HText} from '@/components/atom';
import {EpisodesViewProps} from './types';

const EpisodesView: React.FC<EpisodesViewProps> = ({item}) => {
  const episodes = item?.episode ?? [];

  if (episodes.length == 0) {
    return (
      <HeadingContainer>
        <Cell>
          <HText color="#0C0D34" aligin="center" isBold fontSize="20px">
            No Episodes found
          </HText>
        </Cell>
      </HeadingContainer>
    );
  }
  return (
    <>
      <HeadingContainer>
        <Cell>
          <HText color="#0C0D34" aligin="center" isBold fontSize="20px">
            Episodes
          </HText>
        </Cell>
      </HeadingContainer>
      {episodes?.map((episode, index) => {
        return (
          <Row bgColor={index % 2 == 0 ? '#E8FFFA' : '#FFE4D9'}>
            <Cell>
              <HText color="#0C0D34" aligin="center" fontSize="14px">
                {episode.episode}
              </HText>
            </Cell>
            <Cell>
              <HText color="#0C0D34" aligin="left" fontSize="14px">
                {episode.name}
              </HText>
            </Cell>
          </Row>
        );
      })}
    </>
  );
};

export default EpisodesView;
