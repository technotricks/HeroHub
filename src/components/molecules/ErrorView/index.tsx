import {CenteredVH, HText} from '@/components/atom';
import {SafeScreen} from '@/components/template';

const ErrorView: React.FC = () => {
  return (
    <>
      <SafeScreen>
        <CenteredVH>
          <HText color="#5b5858" aligin="center" isBold fontSize="18px">
            An error occurred. Please try again later.
          </HText>
        </CenteredVH>
      </SafeScreen>
    </>
  );
};

export default ErrorView;
