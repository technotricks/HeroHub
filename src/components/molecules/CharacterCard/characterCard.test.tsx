import 'react-native';
import React from 'react';

import {it, expect, jest, describe} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer, {create, act} from 'react-test-renderer';
import {CharacterCard} from '@/components/molecules';
import {Character} from '@/types/character';

describe('CharacterCard component should render correctly', () => {
  const character: Character = {
    id: 1,
    image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
    name: 'Rick Sanchez',
    status: 'Alive',
    gender: 'Male',
    type: '',
  };

  const mockCallBack = jest.fn();

  const view = create(
    <CharacterCard item={character} onPress={mockCallBack} />,
  );

  it('snapshot', () => {
    expect(view).toMatchSnapshot();
  });

  it('CharacterCard press', () => {
    const button = view.root.findByProps({testID: 'buttonID1'}).props;
    act(() => button.onPress());
    expect(mockCallBack).toBeCalled();
  });

  it('CharacterCard name value render', () => {
    const text = view.root.findByProps({testID: 'nameID1'}).props;
    expect(text).toBeDefined();
  });
});
