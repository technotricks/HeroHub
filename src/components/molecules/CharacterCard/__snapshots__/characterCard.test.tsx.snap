// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard component should render correctly snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "marginBottom": 5,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 5,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="buttonID1"
  >
    <Image
      source={
        {
          "uri": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
        }
      }
      style={
        {
          "borderBottomLeftRadius": 10,
          "borderBottomRightRadius": 10,
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": 100,
          "width": "100%",
        }
      }
      testID="imageID1"
    />
    <Text
      aligin="center"
      color="#FF87A2"
      fontSize="18px"
      isBold={true}
      style={
        {
          "color": "#FF87A2",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      testID="nameID1"
    >
      Rick Sanchez
    </Text>
    <Text
      aligin="left"
      color="#2CB9B0"
      fontSize="16px"
      isBold={true}
      style={
        {
          "color": "#2CB9B0",
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "left",
        }
      }
      testID="statusID1"
    >
      Alive
    </Text>
  </View>
</View>
`;
